src/openforms/submissions/models/submission.py

```diff
@@ -364,9 +324,8 @@ def is_completed(self):
     def remove_sensitive_data(self):
         from .submission_files import SubmissionFileAttachment
 
-        self.bsn = ""
-        self.kvk = ""
-        self.pseudo = ""
+        if hasattr(self, "auth_info"):
+            self.auth_info.clear_sensitive_data()
```

Comment:
perhaps instead of all these `hasattr` checks, we can use `if self.is_authenticated`? That makes the code quite a bit more readable, though it's a bit of a leaky abstraction.